plugin/core/views.py

```diff
@@ -47,65 +50,140 @@
-    (sublime.KIND_ID_TYPE,       "o", "Object",         "meta.mapping"),  # [scope taken from JSON.sublime-syntax]
-    (sublime.KIND_ID_NAVIGATION, "k", "Key",            "meta.mapping.key string"),  # [from JSON.sublime-syntax]
-    (sublime.KIND_ID_VARIABLE,   "n", "Null",           "constant.language"),
-    (sublime.KIND_ID_VARIABLE,   "e", "Enum Member",    "constant.other.enum"),  # Based on {Java,C#}.sublime-syntax
-    (sublime.KIND_ID_TYPE,       "s", "Struct",         "entity.name.struct"),
-    (sublime.KIND_ID_TYPE,       "e", "Event",          "storage.modifier"),   # [scope taken from C#.sublime-syntax]
-    (sublime.KIND_ID_FUNCTION,   "o", "Operator",       "keyword.operator"),
-    (sublime.KIND_ID_TYPE,       "t", "Type Parameter", "storage.type"),
-]
+# sublime.Kind tuples for sublime.CompletionItem, sublime.QuickPanelItem, sublime.ListInputItem
+# https://www.sublimetext.com/docs/api_reference.html#sublime.Kind
+KIND_ARRAY = (sublime.KIND_ID_TYPE, "a", "Array")
+KIND_BOOLEAN = (sublime.KIND_ID_VARIABLE, "b", "Boolean")
+KIND_CLASS = (sublime.KIND_ID_TYPE, "c", "Class")
+KIND_COLOR = (sublime.KIND_ID_MARKUP, "c", "Color")
+KIND_CONSTANT = (sublime.KIND_ID_VARIABLE, "c", "Constant")
+KIND_CONSTRUCTOR = (sublime.KIND_ID_FUNCTION, "c", "Constructor")
+KIND_ENUM = (sublime.KIND_ID_TYPE, "e", "Enum")
+KIND_ENUMMEMBER = (sublime.KIND_ID_VARIABLE, "e", "Enum Member")
+KIND_EVENT = (sublime.KIND_ID_FUNCTION, "e", "Event")
+KIND_FIELD = (sublime.KIND_ID_VARIABLE, "f", "Field")
+KIND_FILE = (sublime.KIND_ID_NAVIGATION, "f", "File")
+KIND_FOLDER = (sublime.KIND_ID_NAVIGATION, "f", "Folder")
+KIND_FUNCTION = (sublime.KIND_ID_FUNCTION, "f", "Function")
+KIND_INTERFACE = (sublime.KIND_ID_TYPE, "i", "Interface")
+KIND_KEY = (sublime.KIND_ID_NAVIGATION, "k", "Key")
+KIND_KEYWORD = (sublime.KIND_ID_KEYWORD, "k", "Keyword")
+KIND_METHOD = (sublime.KIND_ID_FUNCTION, "m", "Method")
+KIND_MODULE = (sublime.KIND_ID_NAMESPACE, "m", "Module")
+KIND_NAMESPACE = (sublime.KIND_ID_NAMESPACE, "n", "Namespace")
+KIND_NULL = (sublime.KIND_ID_VARIABLE, "n", "Null")
+KIND_NUMBER = (sublime.KIND_ID_VARIABLE, "n", "Number")
+KIND_OBJECT = (sublime.KIND_ID_TYPE, "o", "Object")
+KIND_OPERATOR = (sublime.KIND_ID_KEYWORD, "o", "Operator")
+KIND_PACKAGE = (sublime.KIND_ID_NAMESPACE, "p", "Package")
+KIND_PROPERTY = (sublime.KIND_ID_VARIABLE, "p", "Property")
+KIND_REFERENCE = (sublime.KIND_ID_NAVIGATION, "r", "Reference")
+KIND_SNIPPET = (sublime.KIND_ID_SNIPPET, "s", "Snippet")
+KIND_STRING = (sublime.KIND_ID_VARIABLE, "s", "String")
+KIND_STRUCT = (sublime.KIND_ID_TYPE, "s", "Struct")
+KIND_TEXT = (sublime.KIND_ID_MARKUP, "t", "Text")
+KIND_TYPEPARAMETER = (sublime.KIND_ID_TYPE, "t", "Type Parameter")
+KIND_UNIT = (sublime.KIND_ID_VARIABLE, "u", "Unit")
+KIND_VALUE = (sublime.KIND_ID_VARIABLE, "v", "Value")
+KIND_VARIABLE = (sublime.KIND_ID_VARIABLE, "v", "Variable")
+
+COMPLETION_KINDS = {
```

Comment:
So the reason for using a `List` was performance :) a list item lookup is an offset into a contiguous array. But, I guess the performance doesn't matter much. Maybe the dict is even faster, who knows? From what I understand dicts are implemented as hash maps. So a number is run through a hash function before doing an array offset index.

Reply:
I see. As an alternative to the dict, I could use lists and add the fallback `KIND_UNSPECIFIED = (sublime.KIND_ID_AMBIGUOUS, "?", "???")` as a first entry in the list, then indexing with `kind - 1` could also be avoided. But I guess the performance difference (if any) indeed shouldn't matter here and is orders of magnitude smaller than the display refresh time. So it probably doesn't have any impact and would only be a matter of style/preference. And for the lists, there was an additional check `if 1 <= kind <= len(SYMBOL_KINDS)`, which might be even more overhead...